03_12_2020_List/src/telran/OurArrayList.java

```diff
@@ -1,51 +1,56 @@
 package telran;
 
+import java.util.Iterator;
+
 public class OurArrayList<T> implements OurList<T> {
     private static final int INITIAL_CAPACITY = 16;
     private int size;
-    Object[] source;
+    private Object[] source;
 
     public OurArrayList() {
         source = new Object[INITIAL_CAPACITY];
     }
 
     @Override
-    public void addLast(T element) {
+    public void addLast(T element) { // O(n)
+        if (element == null)
+            throw new IndexOutOfBoundsException();
+
         if (size == source.length) {
             increaseCapacity();
         }//source[size++] = element;
         source[size] = element;
         size++;
     }
 
-    void increaseCapacity() {
+    void increaseCapacity() { //O(n)
         int newCapacity = source.length * 2;
         Object[] newSource = new Object[newCapacity];
         System.arraycopy(source, 0, newSource, 0, source.length);
         source = newSource;
     }
 
     @Override
-    public T get(int index) {
+    public T get(int index) { //O(n)
         if (index >= size || index < 0)
             throw new IndexOutOfBoundsException();
         return (T) source[index];
     }
 
     @Override
-    public void set(int index, T value) {
+    public void set(int index, T value) { //O(n)
```

Comment:
here O(1) - since you have a constant number of operations, not depending on the size (n) of the collection